portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -18,76 +18,107 @@
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
 import com.google.common.collect.ImmutableList;
 import com.google.gson.Gson;
 import java.io.IOException;
-import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.sql.Timestamp;
-import java.util.Arrays;
-import java.util.ArrayList;
-import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private final int defaultMaxComments = 15;
+  private final int defaultMaxComments = 15;
+  private final String defaultLanguageCode = "en";
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-        int max_comments;
-        if (request.getParameter("maxComments") == null) {
-            response.sendError(response.SC_BAD_REQUEST, "maxComment parameter missing");
-            max_comments = defaultMaxComments;
-        }
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
-        try{
-            max_comments = Integer.parseInt(request.getParameter("maxComments"));
-        } catch(NumberFormatException e) {
-            response.sendError(response.SC_BAD_REQUEST, "Value entered in \'Set maximum number of comment\' is not a number!");
-            max_comments = defaultMaxComments;
-        }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ImmutableList<Entity> results =
+        ImmutableList.copyOf(
+            datastore
+                .prepare(query)
+                .asList(FetchOptions.Builder.withLimit(getMaxComments(request, response))));
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        ImmutableList<Entity> results = ImmutableList.copyOf(datastore.prepare(query).asList(FetchOptions.Builder.withLimit(max_comments))); 
+    List<Comment> comments = new ArrayList<>();
 
-        List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results) {
+      long id = entity.getKey().getId();
+      String message =
+          getTranslatedComment((String) entity.getProperty("message"), getLanguageCode(request));
+      long timestamp = (long) entity.getProperty("timestamp");
+      Comment comment = new Comment(id, message, timestamp);
```

Comment:
Does this mean the client user will see a timestamp in millis? If that's the case, it's not user friendly though.

Reply:
It's better to convert the timestamp to "date" which is more friendly to users.

Reply:
The client will see the date and time with a clock image before it.
I did the conversion in my script before adding it to my html.
I actually didn't want to send different types of datatypes from Java to JavaScript as it can get a little messy so I just tried to stick to 'long' type.
What do you think?